GitBasic/Lib/Directory/DirectoryStructure.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace GitBasic
+{
+    /// <summary>
+    /// Get Directory Structure class
+    /// Code adapted from https://youtu.be/U2ZvZwDZmJU
+    /// </summary>
+    public static class DirectoryStructure
+    {
+      
+        /// <summary>
+        /// Gets contents of the top level of fullPath
+        /// </summary>
+        /// <param name="fullPath"></param>
+        /// <returns></returns>
+        public static List<DirectoryItem> GetDirectoryContents(string fullPath)

```

Comment:
A couple things about this:

First, don't "try-catch all" - it just hides problems. The only thing I could see going wrong here is the directory not existing. Use an if statement to check for that if you want, or just let it throw.

Second, you can call `Select `on an empty array, and `AddRange `can take an empty collection, so you don't need the if checks `if (dirs.Length > 0)` and `if (fs.Length > 0)`

Finally, regions are best used to group related functions or properties in a class (which you may want to collapse and hide). Best not to use them within a function. I know we don't have an official coding standard for our practicum, so this is more of just a preference - consider removing.

With those changes, it would look something like this:
```
var items = new List<DirectoryItem>();

if (!Directory.Exists(fullPath))
{
      var dirs = Directory.GetDirectories(fullPath);
      items.AddRange(dirs.Select(dir => new DirectoryItem { FullPath = dir, Type = DirectoryItemType.Folder }));

      var fs = Directory.GetFiles(fullPath);
      items.AddRange(fs.Select(file => new DirectoryItem { FullPath = file, Type = DirectoryItemType.File }));
}

return items;
```